lib/elixir/lib/kernel.ex

```diff
@@ -1146,6 +1164,7 @@ defmodule Kernel do
       false
 
   """
+  @spec (arg === arg) :: true when arg: term
```

Comment:
This spec is incorrect, it saying for terms of same type, which are any term, the return is true. Then says same but false. I think previous spec is preferable. Overlapping definitions are awkward and dialyzer wont like them.